oligo_designer_toolsuite/oligo_specificity_filter/_filter_blastn.py

```diff
@@ -187,15 +198,171 @@ def _find_hits(
+                                        table_hits["reference_start_corr"] - table_hits["query_start_corr"],
+                                        table_hits["reference_end_corr"] - (table_hits["query_length"] - table_hits["query_end"]))
+        table_hits["end"] = np.where(table_hits["reference_strand"] == "plus",
+                                        table_hits["reference_end_corr"] + (table_hits["query_length"] - table_hits["query_end"]),
+                                        table_hits["reference_start_corr"] + table_hits["query_start_corr"])
+
+        # create bed file
+        bed = pd.DataFrame({  
+            "chr": table_hits["reference"],  
+            "start": table_hits["start"],  
+            "end": table_hits["end"],  
+            "name": table_hits["query"],  
+            "score": 0,  
+            "strand": table_hits["reference_strand"].map({"plus": "+", "minus": "-"})  
+        })
+        # adjust for possible overflows (e.g. new coordinates are not included in the gene boundaries)
+        # additionally we store how muchpadding we have to do to have two seqeunces of the same length
+        bed["overflow_start"] = bed["start"].apply(lambda x: -x if x < 0 else 0)
+        bed["start"] = bed["start"].apply(lambda x: x if x >= 0 else 0)
+        # what about having the leght of a region as a method of the reference database?
+        file_reference = reference_database.write_database_to_fasta(
+            filename="reference_db"
+        )
+        records = SeqIO.index(file_reference, "fasta")
+        bed["len_region"] = bed["chr"].apply(lambda x: len(records[x].seq))
+        bed["overflow_end"] = bed[["end", "len_region"]].apply(
+            lambda x: x["end"] - x["len_region"] if x["end"] > x["len_region"] else 0,
+            axis=1,
+        )
+        bed["end"] = bed[["end", "len_region"]].apply(
+            lambda x: x["end"] if x["end"] <= x["len_region"] else x["len_region"],
+            axis=1,
+        )
+        bed.sort_index(inplace=True)
+        file_bed = os.path.join(self.dir_output, f"references_{region_id}.bed")
+        bed.to_csv(
+            file_bed,
+            sep="\t",
+            index=False,
+            header=False,
+            columns=["chr", "start", "end", "name", "score", "strand"],
+        )
+
+        # generate the fasta file
+        references_fasta_file = os.path.join(
+            self.dir_output, f"references_{region_id}.fasta"
+        )
+        get_sequence_from_annotation(
+            file_bed, file_reference, references_fasta_file, strand=True, nameOnly=True
+        )
+
+        references = [
+            off_reference.seq for off_reference in SeqIO.parse(references_fasta_file, "fasta")
+        ]
+        references_padded = []
+        for reference, overflow_start, overflow_end, strand in zip(
+            references, bed["overflow_start"], bed["overflow_end"], bed["strand"]
+        ):
+            if overflow_start != 0:
```

Comment:
I think this would be more readable:
```
padding_start = "-" * overflow_start if overflow_start != 0 else ""
padding_end = "-" * overflow_end if overflow_end != 0 else ""

if strand == "+":
    reference = padding_start + reference + padding_end
elif strand == "-":
    reference = padding_end + reference + padding_start
    
```